### ISSUE TITLE
ENH: add "right" argument to "qcut", like "cut" has

### ISSUE THREAD

**Issue by dwayne298** [pandas-dev/pandas#58933]
### Feature Type

- [X] Adding new functionality to pandas

- [ ] Changing existing functionality in pandas

- [ ] Removing existing functionality in pandas

### Problem Description

In the `cut` function, you can specify "right" which defines whether `bins` includes the rightmose edge or not. It would be useful to have the same functionality with `qcut`

### Feature Description

From a naïve perspective, I think you could just add `right=right` when `_bins_to_cuts` is called, just as it is done in `cut`

### Alternative Solutions

Current alternative I think would be to calculate the quantiles outside of pandas and then use those as `bins` for `cut`

**Comment by aaronchucarroll** [comment]
take

**Comment by giormala** [comment]
take

**Comment by ucheetah** [comment]
take

**Comment by rhshadrach** [comment]
Labeled as a closing candidate as in https://github.com/pandas-dev/pandas/issues/62938#issuecomment-4760144312

**Comment by rhshadrach** [comment]
The workaround in https://github.com/pandas-dev/pandas/pull/63053#issuecomment-4760142981 is not a drop-in replacement (as was suggested may be the case there), it drops the maximum value.

```python
s = pd.Series([1, 2, 3, 4, 5])
pd.cut(s, bins=s.quantile(np.linspace(0, 1, 3)), right=False).tolist()
# [[1.0, 3.0), [1.0, 3.0), [3.0, 5.0), [3.0, 5.0), nan]
```

I'm positive on this change and have removed the Closing Candidate label. @jbrockmendel - reinstate if you disagree.

**Comment by jbrockmendel** [comment]
Nope, if the workaround isn’t a drop-in then you’re right to de-label.
